internal/annotation/annotations.go

```diff
@@ -184,15 +187,15 @@ func NumRetries(i *v1beta1.Ingress) uint32 {
 
 // PerTryTimeout returns the duration envoy will wait per retry cycle.
 func PerTryTimeout(i *v1beta1.Ingress) time.Duration {
```

Comment:
Do you want to change these to `metav1.ObjectMetaAccessor` as well in this PR, or is that too much?

Reply:
That seems doable, for these specific Ingress ones. I don't want to touch `dag` yet though.

Reply:
after looking at this, it's more substantial than I thought - I'll hold off to another PR.